### ISSUE TITLE
SNS does not use DEFAULT_REGION

### ISSUE THREAD

**Issue by deganmee** [localstack/localstack#731]
I am using localstack with docker.  I have specified DEFAULT_REGION=us-east-1 in docker-compose.yml:
```
version: '2.1'

services:
  localstack:
    image: localstack/localstack
    ports:
      - "4567-4583:4567-4583"
      - "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - SERVICES=sqs,sns,s3
      - DEFAULT_REGION=us-east-1
```

When I create an SNS topic from the command line, the ARN shows that the topic has been created in us-west-2:
```
 $ aws --endpoint-url=http://localhost:4575 sns create-topic --name my-topic
{
    "TopicArn": "arn:aws:sns:us-west-2:123456789012:my-topic"
}
```

I have worked around the issue by specifying --region on the aws command line:
```
$ aws --endpoint-url=http://localhost:4575 sns create-topic --name my-topic --region us-east-1
{
    "TopicArn": "arn:aws:sns:us-east-1:123456789012:my-topic"
}
```

I believe that SNS should adopt DEFAULT_REGION that is specified in docker-compose.yml like the other services do.

Another issue/question related to SNS in localstack - I am not seeing my SNS topics appear in the localstack dashboard.  I expect to see the topics that have been created in SNS.

**Comment by amitdhawan** [comment]
Any updates on this issue?

**Comment by tillkuhn** [comment]
Same here, neither DEFAULT_REGION nor AWS_DEFAULT_REGION make a difference. Localstack version 0.8.6 on MacOS (without docker).

**Comment by bensoille** [comment]
Hi there

Same issue here, with localstack 0.9.1 on linux, with docker.
In addition, the pasted account id (visible in arn) is wrong (123456789012 instead of 000000000000) compared to other services.

Thx in advance !!

**Comment by whummer** [comment]
Thanks for reporting @deganmee @amitdhawan @tillkuhn @bensoille . This should be fixed in #1744 - it should now work with a combination of `DEFAULT_REGION` and `AWS_DEFAULT_REGION`. (`AWS_DEFAULT_REGION` takes precedence in the current configuration setup)

Please note that your testing code should not rely too heavily on server-side `DEFAULT_REGION` but also configure the region properly on the client side. Future releases of LocalStack may remove the concept of `DEFAULT_REGION` and focus on supporting the region provided by the client (which is closer to how the real AWS environment operates).

**Comment by Ghazgkull** [comment]
@bensoille @whummer I'm still seeing the issue mentioned in the comment above (https://github.com/localstack/localstack/issues/731#issuecomment-491337055), about the ARN of SNS topics being generated with an account number of "123456789012". This is causing Terraform to blow up as it's expecting the account number to be "000000000000". Is that issue tracked anywhere?
